internal/datasource/git.go

```diff
@@ -31,20 +36,66 @@ func (s *GitSource) ProjectSource(body hcl.Body, ctx *hcl.EvalContext) (*pb.Job_
 		return nil, diag
 	}
 
+	// Start building the result
+	result := &pb.Job_Git{
+		Url:  cfg.Url,
+		Path: cfg.Path,
+	}
+	switch {
+	case cfg.Username != "":
+		result.Auth = &pb.Job_Git_Basic_{
+			Basic: &pb.Job_Git_Basic{
+				Username: cfg.Username,
+				Password: cfg.Password,
+			},
+		}
+
+	case cfg.SSHKey != "":
+		result.Auth = &pb.Job_Git_Ssh{
```

Comment:
I think here we should do a little error checking that the given SSHKey can be decrypted using the given password. Because the usage of the key happens very far from the configuration, it will be hard for them to realize they used the wrong password (or no password at all) when we try to use the ssh key, but much easier if we check here and tell them.

Reply:
Yes this is a very good idea.